extension/encoding/awslogsencodingextension/unmarshaler/subscription-filter/subscription_filter.go

```diff
@@ -0,0 +1,140 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package subscriptionfilter // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/unmarshaler/subscription-filter"
+
+import (
+	"bytes"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-lambda-go/events"
+	jsoniter "github.com/json-iterator/go"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.27.0"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/internal/metadata"
+)
+
+var (
+	errEmptyOwner     = errors.New("cloudwatch log with message type 'DATA_MESSAGE' has empty owner field")
+	errEmptyLogGroup  = errors.New("cloudwatch log with message type 'DATA_MESSAGE' has empty log group field")
+	errEmptyLogStream = errors.New("cloudwatch log with message type 'DATA_MESSAGE' has empty log stream field")
+)
+
+func validateLog(log events.CloudwatchLogsData) error {
+	switch log.MessageType {
+	case "DATA_MESSAGE":
+		if log.Owner == "" {
+			return errEmptyOwner
+		}
+		if log.LogGroup == "" {
+			return errEmptyLogGroup
+		}
+		if log.LogStream == "" {
+			return errEmptyLogStream
+		}
+	case "CONTROL_MESSAGE":
+	default:
+		return fmt.Errorf("cloudwatch log has invalid message type %q", log.MessageType)
+	}
+	return nil
+}
+
+type subscriptionFilterUnmarshaler struct {
+	buildInfo component.BuildInfo
+
+	// We need to decompress the records.
+	// Creating a new gzip.Reader every time is expensive.
+	// We can keep the gzip in cache, and take advantage
+	// of the continuously AWS environment or the AWS
+	// warm starts, in case of lambda.
+	// See https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtime-environment.html.
```

Comment:
```suggestion
	// Pool the gzip readers, which are expensive to create.
```

I don't think we need all that information here, let's keep it simple.

For performance reasons we should consider using klauspost/compress here, like in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6ce59264931a9e348b6d483748aef5d38c801342/receiver/awsfirehosereceiver/internal/unmarshaler/cwlog/unmarshaler.go#L15

Reply:
Yes, I've switched.